flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkWriter.java

```diff
@@ -56,6 +61,31 @@
 class KinesisDataStreamsSinkWriter<InputT> extends AsyncSinkWriter<InputT, PutRecordsRequestEntry> {
     private static final Logger LOG = LoggerFactory.getLogger(KinesisDataStreamsSinkWriter.class);
 
+    private static final RetryableExceptionClassifier RESOURCE_NOT_FOUND_STRATEGY =
+            RetryableExceptionClassifier.withRootCauseOfType(
+                    ResourceNotFoundException.class,
+                    err ->
+                            new KinesisDataStreamsException(
+                                    "Encountered non-recoverable exception relating to not being able to find the specified resources",
+                                    err));
+
+    private static final RetryableExceptionClassifier NON_RECOVERABLE_EXCEPTION_STRATEGY =
+            RetryableExceptionClassifier.withRootCauseOfType(
+                    Error.class,
+                    err ->
+                            new KinesisDataStreamsException(
+                                    "Encountered non-recoverable exception in the Kinesis Data Streams Sink",
+                                    err));
+
+    private static final RetryableExceptionClassifier KINESIS_RETRY_VALIDATION_STRATEGY =
+            RetryableExceptionClassifier.createChain(
+                    GENERAL_ERROR_STRATEGY.clone(),
```

Comment:
Cloning a singleton instance is an odd way to achieve this. Factories with builder methods or Builders are a more standard approach. For instance `AsyncSinkThrowableWrappers.generalErrorStrategy()`, can we please refactor this?